### ISSUE TITLE
Fix Storybook compilation warnings

### ISSUE THREAD

**Issue by stefanalund** [spotify/backstage#700]
## Expected Behavior

Don't show errors or warnings.

## Current Behavior

When you run `yarn start` inside `packages/storybook` you get this:

<img width="1105" alt="Screenshot 2020-05-02 at 15 22 07" src="https://user-images.githubusercontent.com/190856/80865319-d24e2900-8c88-11ea-98e2-8ffbc5dff86f.png">

## Steps to Reproduce

1. Step 1
2. Step 2
3. ...

## Your Environment

- NodeJS Version (v12):
- Operating System and Version (e.g. Ubuntu 14.04):
- Browser Information:

**Comment by OrkoHunter** [comment]
All these warnings are strange, and the cause is same. It is happening when we are exporting a `type` i.e. `export type Props` from a module and then exporting it further from the layout/component's `index.ts`.

Very curious issue. :)

**Comment by stefanalund** [comment]
Closed in #701

**Comment by OrkoHunter** [comment]
There is a new warning from #600, which is similar to these, but not exactly the same. `TableCoulmn` is an `interface` instead of a `type`.

```

WARNING in ../core/src/components/Table/index.ts 1:0-47
"export 'TableColumn' was not found in './Table'
 @ ../core/src/components/Table/Table.stories.tsx
 @ ../core/src/components sync ^\.\/(?:(?:(?!\.)(?:(?:(?!(?:|\/)\.).)*?)\/)?(?!\.)(?=.)[^/]*?\.stories\.tsx\/?)$
 @ ./.storybook/generated-entry.js
 @ multi /Users/himanshu/workspace/spotify/backstage/node_modules/@storybook/react/node_modules/@storybook/core/dist/server/common/polyfills.js /Users/himanshu/workspace/spotify/backstage/node_modules/@storybook/react/node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/config.js ./.storybook/generated-entry.js (webpack)-hot-middleware/client.js?reload=true&quiet=true
```
